### PR TITLE
macos: fix linker error from CEL library

### LINKED ISSUE #31932: Envoy v1.29.0 build fails on Mac OS
*Title*: *Envoy v1.29.0 build fails on Mac OS*

*Description*:
When I am building Envoy v1.29.0 on MacOS 12.7.1 with clang 14.0.0 I noticed an error

```
ec2-user@Mac-mini envoy-builds % gcc --version
Apple clang version 14.0.0 (clang-1400.0.29.202)
Target: x86_64-apple-darwin21.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

Envoy build error

```
Undefined symbols for architecture x86_64:
  "cel::interop_internal::ProtoStructValueToMessageWrapper(cel::Value const&)", referenced from:
      cel::interop_internal::ToLegacyValue(google::protobuf::Arena*, cel::Handle<cel::Value> const&, bool) in libinterop.a(interop.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Target //contrib/exe:envoy-static failed to build
INFO: Elapsed time: 136.522s, Critical Path: 46.32s
```

I've tried both conrtib and not contrib build but I have the same issue.

*Repro steps*:

1. Use mac 12.7.1 or 11.7.10 (on aws ami-06bf601fad285cc69 or ami-0da052d2225304b8f)
2. Setup CI tools
3. run bazel build -c opt envoy
4. Notice error at the last step

```
Undefined symbols for architecture x86_64:
  "cel::interop_internal::ProtoStructValueToMessageWrapper(cel::Value const&)", referenced from:
      cel::interop_internal::ToLegacyValue(google::protobuf::Arena*, cel::Handle<cel::Value> const&, bool) in libinterop.a(interop.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Target //contrib/exe:envoy-static failed to build
INFO: Elapsed time: 136.522s, Critical Path: 46.32s
```

I've noticed that more people have the same issue https://github.com/Homebrew/homebrew-core/pull/160068

### PR DESCRIPTION
**Pull request by ggreenway**
Commit Message:
Additional Description: As described in the linked issue, on macOS (at least in some OS/compiler combos) there is a linker error in CEL. The missing symbol is specified as a weak symbol. The definition of the function is in `cel-cpp/extensions/protobuf:data`. As far as I can tell, only a test in CEL actually builds that target; Envoy does not build that target, so this function will not ever be compiled into Envoy. So commenting out the block in this PR should not have behavior change to how Envoy uses CEL.
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
Fixes #31932
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by juanmolle — COMMENTED** [review]
LGTM, I check the patch and works in:
MacOS Sonoma 14.2.1 
Apple clang version 15.0.0 (clang-1500.1.0.2.5)
Target: arm64-apple-darwin23.2.0

**Comment by moderation** [comment]
Confirmed this also works for me. Fantastic. No impact on my Linux builds. LGTM

**Comment by htuch** [comment]
Not super excited carrying a permanent patch for this. Can we just supply the symbol conditioned in this test or link to some definition there? That also won't affect Envoy production builds and seems more maintainable.

**Comment by yanavlasov** [comment]
> Not super excited carrying a permanent patch for this for maintainability reasons, it makes upgrades for dep shepherds much harder.
> 
> Why don't we just link in the target you mention to the CEL library dependencies via BUILD?

If we link a missing file it will still be a patch, just to a BUILD file. And also unknown behavior change in prod code.
I will raise this issue to the CEL maintainers . There is an OSS issue filed for this in their repo https://github.com/google/cel-cpp/issues/205 but it does not seem to get much traction.

**Review by yanavlasov — APPROVED** [review]

**Comment by htuch** [comment]
> If we link a missing file it will still be a patch, just to a BUILD file. And also unknown behavior change in prod code.

I'm not suggesting this. I'm suggesting to link to the actual definition. But I don't even see this in https://github.com/google/cel-cpp/blob/master/extensions/protobuf. @ggreenway can you share a link to the ref?

**Comment by ggreenway** [comment]
I attempted to update CEL, but the version of CEL with the needed change depends on a feature in `absl` that wasn't added until `2023-11-08`, but Envoy is on absl `20230802.1`. I think we should go ahead with this fix for now, backport it to 1.29, and then someone can eventually upgrade CEL and absl and delete the extra patch being added here.

**Comment by ramaraochavali** [comment]
@htuch @ggreenway Can we please merge this and can be removed later when we upgrade CEL?

**Comment by htuch** [comment]
@ggreenway can you file an issue to followup to remove the patch? 

In the absence of @kyessenov maybe @TristonianJones can help verify the safety here?

**Comment by ggreenway** [comment]
Filed #32261

**Comment by jcking** [comment]
This should be "relatively" safe so long as you are still using everything under `google::api::expr::runtime`.

**Review by htuch — APPROVED** [review]

**Pull request merged**
